### PR TITLE
[MM-15268] Add a config.json setting for minimum username length 

### PR DESCRIPTION
**Pull request by carmo-evan**
#### Summary

[MM-15268] Add a config.json setting for minimum username length 

#### Ticket Link

https://github.com/mattermost/mattermost-server/issues/10702

### Client-Side Changes

https://github.com/mattermost/mattermost-webapp/pull/2895

[MM-15268]: https://mattermost.atlassian.net/browse/MM-15268

### CODE REVIEW

**Comment by carmo-evan** [comment]
@lieut-data ready for review.

**Comment by hanzei** [comment]
@crspeller @levb Gentle reminder about this PR.

**Comment by hanzei** [comment]
@carmo-evan Would you please merger `master` into your branch?

File: app/authentication.go [hunk 1]
```diff
@@ -45,6 +45,15 @@ func (a *App) IsPasswordValid(password string) *model.AppError {
 	return utils.IsPasswordValidWithSettings(password, &a.Config().PasswordSettings)
 }
 
+// IsUsernameValid adds extra checks beyond those found in model/user, particularly, those dependent on config values.
+func (a *App) IsUsernameValid(username string) *model.AppError {
```
Comment (levb) [inline app/authentication.go]: 0/5 suggest keeping it private for now, `isUsernameValie()`

File: model/config.go [hunk 2]
```diff
@@ -844,6 +849,8 @@ type SqlSettings struct {
 	Trace                       *bool    `restricted:"true"`
 	AtRestEncryptKey            *string  `restricted:"true"`
 	QueryTimeout                *int     `restricted:"true"`
+	InnodbFtMinTokenSize        *int     `restricted:"true"`
```
Comment (levb) [inline model/config.go]: ??? Why are these here

File: store/storetest/settings.go [hunk 3]
```diff
@@ -139,11 +139,15 @@ func databaseSettings(driver, dataSource string) *model.SqlSettings {
 		Trace:                       model.NewBool(false),
 		AtRestEncryptKey:            model.NewString(model.NewRandomString(32)),
 		QueryTimeout:                new(int),
+		InnodbFtMinTokenSize:        new(int),
```
Comment (levb) [inline store/storetest/settings.go]: ditto

**Comment by hanzei** [comment]
Closing this one due to inactivity. @RajatVaryani will now work on this topic.

File: model/config_test.go [hunk 4]
```diff
@@ -26,6 +26,12 @@ func TestConfigDefaults(t *testing.T) {
 		require.True(t, checkNowhereNil(t, "config", c))
 	})
 
+	t.Run("MinimumUsernameLength not nil when initialized",func(t *testing.T){
+		c := Config{}
+		c.SetDefaults()
+		require.NotNil(t,c.ServiceSettings.MinimumUsernameLength)
+	})
```
Comment (leite08) [inline model/config_test.go]: Maybe also test if `MinimumUsernameLength` is not overridden if initialized?
